integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/EnvironmentChildTest.java

```diff
@@ -231,4 +238,21 @@ private boolean parentNameEquals(EnvironmentTestDto environment, SimpleEnvironme
                                 .bodyContains(method)
                                 .exactTimes(times));
     }
+
+    private Assertion<EnvironmentTestDto, EnvironmentClient> checkEnvsAreNotListedByName(List<String> environmentNames) {
+        return (testContext, environmentTestDto, environmentClient) -> {
+            Collection<SimpleEnvironmentResponse> simpleEnvironmentV4Respons = environmentTestDto.getResponseSimpleEnvSet();
+            if (isNull(simpleEnvironmentV4Respons)) {
+                throw new TestFailException("Environment list response is missing.");
+            }
+            List<String> listedEnvironmentNames = simpleEnvironmentV4Respons.stream()
+                    .filter(response -> !response.getEnvironmentStatus().equals(EnvironmentStatus.ARCHIVED))
+                    .map(EnvironmentBaseResponse::getName)
+                    .collect(Collectors.toList());
+            if (listedEnvironmentNames.stream().anyMatch(environmentNames::contains)) {
+                throw new TestFailException("Environment is listed");
```

Comment:
From the message we can not tell which env was listed - the parent or the child.
I have also refactored this test case in my PR, see here: https://github.com/hortonworks/cloudbreak/pull/8923/files#diff-68f3ed9be95a1bffc91d2b7ec253c58c
I think we should merge these somehow here

Reply:
I think we should merge your PR first, after that rebase this PR onto master with your changes...